owlkettle/guidsl.nim

```diff
@@ -93,6 +94,35 @@ proc parseGui(node: NimNode): Node =
         result = node[0].parseGui()
       for it in 1..<node.len:
         result.children.add(node[it].parseGui())
+    of nnkInfix: # For expressions like "<Widget> as <stateRefVariable>"
+      let isRefAssignmentExpression = node[0].eqIdent("as")
+      if not isRefAssignmentExpression:
+        error("You can only use infix for assigning stateReferences. That must be done via '<Widget> as <stateRefVariable>' syntax")
+      let widgetName = case node[1].kind:
+        of nnkIdent: node[1]
+        of nnkCall: node[1][0]
+        else: 
+          error("Tried to use 'as' with invalid syntax", node)
+          newEmptyNode() # Forces the compiler to acknowlege that all branches of the case statement return a NimNode
+      
+      let stateRefVar = case node[2].kind:
+        of nnkPragmaExpr: node[2][0]
+        else: node[2]
```

Comment:
This just ignores the adder, doesn't it?

If you want to use the syntax `Widget {.adder.} as ...`. Please report an error if the user tries to use `Widget as ... {.adder.}`.

Reply:
It does not in fact. Like, I didn't think deeper about that fact, but the DSL combination that gets created by `Widget {.somepragma.}  as ...` appears to run into the else block in line 105.
```nim
        else: 
          error("Tried to use 'as' with invalid syntax", node)
```

At least when I added
```nim          
          ListBoxRow() {.addRow.} as app.dummyRef:
            Label(text = "Static last")
```
To the search_entry example it blew up in my face with exactly that error message.